Unable to open dev options

I today updated to 314.0.0.5.114 beta because i saw that i needed to get beta in issue thread. What should my class nams, method name and other class name be (im kinda confused what to do)

```
[ 2024-01-20T12:57:27.815    10375: 24814: 24814 I/LSPosed         ] Loading xposed for com.chacha.igexperiments/10375
[ 2024-01-20T12:57:27.851    10375: 24814: 24814 I/LSPosed-Bridge  ] Loading legacy module com.chacha.igexperiments from /data/app/~~cMT-C-vvE1l7dtPcVCDWfg==/com.chacha.igexperiments-GTgEjI_nxLDATYi42Gt-iQ==/base.apk
[ 2024-01-20T12:57:28.015    10375: 24814: 24814 I/LSPosed-Bridge  ]   Loading class com.chacha.igexperiments.Module
[ 2024-01-20T12:57:28.966    10375: 24814: 24814 W/LSPosed         ] New modules detected, hook preferences
[ 2024-01-20T12:57:43.564     2000: 29730: 29730 I/LSPosed         ] Loaded manager, skipping next steps
[ 2024-01-20T12:57:48.235        0:  4346:  4363 I/lspd            ] Background concurrent copying GC freed 344659(23MB) AllocSpace objects, 12(2608KB) LOS objects, 83% free, 4779KB/28MB, paused 78us,72us total 144.304ms
[ 2024-01-20T12:57:54.955        0:  4346:  4372 I/LSPosedLogcat   ] !!refresh_modules!!
[ 2024-01-20T12:57:54.962        0:  4346:  4374 I/LSPosedLogcat   ] New log file: /data/adb/lspd/log/modules_2024-01-20T11:57:54.959.log
[ 2024-01-20T12:58:09.045    10250: 29731: 29731 I/LSPosed         ] Loading xposed for com.instagram.android/10250
[ 2024-01-20T12:58:09.066    10250: 29731: 29731 I/LSPosed-Bridge  ] Loading legacy module com.chacha.igexperiments from /data/app/~~cMT-C-vvE1l7dtPcVCDWfg==/com.chacha.igexperiments-GTgEjI_nxLDATYi42Gt-iQ==/base.apk
[ 2024-01-20T12:58:09.268    10250: 29731: 29731 I/LSPosed-Bridge  ]   Loading class com.chacha.igexperiments.Module
[ 2024-01-20T12:58:09.701    10250: 29731: 29731 I/LSPosed-Bridge  ] (IGExperiments) Using class name from Github
[ 2024-01-20T12:58:09.763    10250: 29731: 29731 E/LSPosed-Bridge  ] TgVlQGkRWEF.clsVYnC.Bz.XposedHelpers$InvocationTargetError: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /data/app/~~SJg0T_IaHTZQOUzHwOP5nw==/com.instagram.android-_4dHFA7-NyrnB4fWnTJ_kg==/base.apk
	at TgVlQGkRWEF.clsVYnC.Bz.XposedHelpers.callMethod(SourceFile:2)
	at com.chacha.igexperiments.Module.handleLoadPackage(Module.java:115)
	at TgVlQGkRWEF.clsVYnC.Bz.IXposedHookLoadPackage$Wrapper.handleLoadPackage(Unknown Source:2)
	at TgVlQGkRWEF.clsVYnC.Bz.callbacks.XC_LoadPackage.call(Unknown Source:6)
	at TgVlQGkRWEF.clsVYnC.Bz.callbacks.XCallback.callAll(Unknown Source:26)
	at h0.a(Unknown Source:271)
	at java.lang.reflect.Method.invoke(Native Method)
	at J.callback(Unknown Source:250)
	at LSPHooker_.getClassLoader(Unknown Source:8)
	at android.app.LoadedApk.getResources(LoadedApk.java:1310)
	at android.app.ContextImpl.createAppContext(ContextImpl.java:3014)
	at android.app.ContextImpl.createAppContext(ContextImpl.java:3006)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6623)
	at android.app.ActivityThread.access$1500(ActivityThread.java:247)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2053)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7839)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
Caused by: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /data/app/~~SJg0T_IaHTZQOUzHwOP5nw==/com.instagram.android-_4dHFA7-NyrnB4fWnTJ_kg==/base.apk
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1231)
	at android.content.pm.PackageParser.parseMonolithicPackage(PackageParser.java:1177)
	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1081)
	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1090)
	at java.lang.reflect.Method.invoke(Native Method)
	... 22 more
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.app.Application.getSystemService(java.lang.Class)' on a null object reference
	at android.content.pm.PackageParser.parseBaseApkCommon(PackageParser.java:2400)
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1892)
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1215)
	... 26 more
[ 2024-01-20T12:58:16.584    10250: 31784: 31784 I/LSPosed         ] Loading xposed for com.instagram.android:mqtt/10250
[ 2024-01-20T12:58:16.634    10250: 31784: 31784 I/LSPosed-Bridge  ] Loading legacy module com.chacha.igexperiments from /data/app/~~cMT-C-vvE1l7dtPcVCDWfg==/com.chacha.igexperiments-GTgEjI_nxLDATYi42Gt-iQ==/base.apk
[ 2024-01-20T12:58:16.991    10250: 31784: 31784 I/LSPosed-Bridge  ]   Loading class com.chacha.igexperiments.Module
[ 2024-01-20T12:58:17.355    10250: 31784: 31784 I/LSPosed-Bridge  ] (IGExperiments) Using class name from Github
[ 2024-01-20T12:58:17.497    10250: 31784: 31784 E/LSPosed-Bridge  ] TgVlQGkRWEF.clsVYnC.Bz.XposedHelpers$InvocationTargetError: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /data/app/~~SJg0T_IaHTZQOUzHwOP5nw==/com.instagram.android-_4dHFA7-NyrnB4fWnTJ_kg==/base.apk
	at TgVlQGkRWEF.clsVYnC.Bz.XposedHelpers.callMethod(SourceFile:2)
	at com.chacha.igexperiments.Module.handleLoadPackage(Module.java:115)
	at TgVlQGkRWEF.clsVYnC.Bz.IXposedHookLoadPackage$Wrapper.handleLoadPackage(Unknown Source:2)
	at TgVlQGkRWEF.clsVYnC.Bz.callbacks.XC_LoadPackage.call(Unknown Source:6)
	at TgVlQGkRWEF.clsVYnC.Bz.callbacks.XCallback.callAll(Unknown Source:26)
	at h0.a(Unknown Source:271)
	at java.lang.reflect.Method.invoke(Native Method)
	at J.callback(Unknown Source:250)
	at LSPHooker_.getClassLoader(Unknown Source:8)
	at android.app.LoadedApk.getResources(LoadedApk.java:1310)
	at android.app.ContextImpl.createAppContext(ContextImpl.java:3014)
	at android.app.ContextImpl.createAppContext(ContextImpl.java:3006)
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6623)
	at android.app.ActivityThread.access$1500(ActivityThread.java:247)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2053)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7839)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
Caused by: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /data/app/~~SJg0T_IaHTZQOUzHwOP5nw==/com.instagram.android-_4dHFA7-NyrnB4fWnTJ_kg==/base.apk
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1231)
	at android.content.pm.PackageParser.parseMonolithicPackage(PackageParser.java:1177)
	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1081)
	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1090)
	at java.lang.reflect.Method.invoke(Native Method)
	... 22 more
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.app.Application.getSystemService(java.lang.Class)' on a null object reference
	at android.content.pm.PackageParser.parseBaseApkCommon(PackageParser.java:2400)
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1892)
	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1215)
	... 26 more
```

here i succeed copying from log file i hope that its enough

> ```
> [ 2024-01-20T12:57:27.815    10375: 24814: 24814 I/LSPosed         ] Loading xposed for com.chacha.igexperiments/10375
> [ 2024-01-20T12:57:27.851    10375: 24814: 24814 I/LSPosed-Bridge  ] Loading legacy module com.chacha.igexperiments from /data/app/~~cMT-C-vvE1l7dtPcVCDWfg==/com.chacha.igexperiments-GTgEjI_nxLDATYi42Gt-iQ==/base.apk
> [ 2024-01-20T12:57:28.015    10375: 24814: 24814 I/LSPosed-Bridge  ]   Loading class com.chacha.igexperiments.Module
> [ 2024-01-20T12:57:28.966    10375: 24814: 24814 W/LSPosed         ] New modules detected, hook preferences
> [ 2024-01-20T12:57:43.564     2000: 29730: 29730 I/LSPosed         ] Loaded manager, skipping next steps
> [ 2024-01-20T12:57:48.235        0:  4346:  4363 I/lspd            ] Background concurrent copying GC freed 344659(23MB) AllocSpace objects, 12(2608KB) LOS objects, 83% free, 4779KB/28MB, paused 78us,72us total 144.304ms
> [ 2024-01-20T12:57:54.955        0:  4346:  4372 I/LSPosedLogcat   ] !!refresh_modules!!
> [ 2024-01-20T12:57:54.962        0:  4346:  4374 I/LSPosedLogcat   ] New log file: /data/adb/lspd/log/modules_2024-01-20T11:57:54.959.log
> [ 2024-01-20T12:58:09.045    10250: 29731: 29731 I/LSPosed         ] Loading xposed for com.instagram.android/10250
> [ 2024-01-20T12:58:09.066    10250: 29731: 29731 I/LSPosed-Bridge  ] Loading legacy module com.chacha.igexperiments from /data/app/~~cMT-C-vvE1l7dtPcVCDWfg==/com.chacha.igexperiments-GTgEjI_nxLDATYi42Gt-iQ==/base.apk
> [ 2024-01-20T12:58:09.268    10250: 29731: 29731 I/LSPosed-Bridge  ]   Loading class com.chacha.igexperiments.Module
> [ 2024-01-20T12:58:09.701    10250: 29731: 29731 I/LSPosed-Bridge  ] (IGExperiments) Using class name from Github
> [ 2024-01-20T12:58:09.763    10250: 29731: 29731 E/LSPosed-Bridge  ] TgVlQGkRWEF.clsVYnC.Bz.XposedHelpers$InvocationTargetError: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /data/app/~~SJg0T_IaHTZQOUzHwOP5nw==/com.instagram.android-_4dHFA7-NyrnB4fWnTJ_kg==/base.apk
> 	at TgVlQGkRWEF.clsVYnC.Bz.XposedHelpers.callMethod(SourceFile:2)
> 	at com.chacha.igexperiments.Module.handleLoadPackage(Module.java:115)
> 	at TgVlQGkRWEF.clsVYnC.Bz.IXposedHookLoadPackage$Wrapper.handleLoadPackage(Unknown Source:2)
> 	at TgVlQGkRWEF.clsVYnC.Bz.callbacks.XC_LoadPackage.call(Unknown Source:6)
> 	at TgVlQGkRWEF.clsVYnC.Bz.callbacks.XCallback.callAll(Unknown Source:26)
> 	at h0.a(Unknown Source:271)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at J.callback(Unknown Source:250)
> 	at LSPHooker_.getClassLoader(Unknown Source:8)
> 	at android.app.LoadedApk.getResources(LoadedApk.java:1310)
> 	at android.app.ContextImpl.createAppContext(ContextImpl.java:3014)
> 	at android.app.ContextImpl.createAppContext(ContextImpl.java:3006)
> 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6623)
> 	at android.app.ActivityThread.access$1500(ActivityThread.java:247)
> 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2053)
> 	at android.os.Handler.dispatchMessage(Handler.java:106)
> 	at android.os.Looper.loopOnce(Looper.java:201)
> 	at android.os.Looper.loop(Looper.java:288)
> 	at android.app.ActivityThread.main(ActivityThread.java:7839)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
> 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
> Caused by: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /data/app/~~SJg0T_IaHTZQOUzHwOP5nw==/com.instagram.android-_4dHFA7-NyrnB4fWnTJ_kg==/base.apk
> 	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1231)
> 	at android.content.pm.PackageParser.parseMonolithicPackage(PackageParser.java:1177)
> 	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1081)
> 	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1090)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	... 22 more
> Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.app.Application.getSystemService(java.lang.Class)' on a null object reference
> 	at android.content.pm.PackageParser.parseBaseApkCommon(PackageParser.java:2400)
> 	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1892)
> 	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1215)
> 	... 26 more
> [ 2024-01-20T12:58:16.584    10250: 31784: 31784 I/LSPosed         ] Loading xposed for com.instagram.android:mqtt/10250
> [ 2024-01-20T12:58:16.634    10250: 31784: 31784 I/LSPosed-Bridge  ] Loading legacy module com.chacha.igexperiments from /data/app/~~cMT-C-vvE1l7dtPcVCDWfg==/com.chacha.igexperiments-GTgEjI_nxLDATYi42Gt-iQ==/base.apk
> [ 2024-01-20T12:58:16.991    10250: 31784: 31784 I/LSPosed-Bridge  ]   Loading class com.chacha.igexperiments.Module
> [ 2024-01-20T12:58:17.355    10250: 31784: 31784 I/LSPosed-Bridge  ] (IGExperiments) Using class name from Github
> [ 2024-01-20T12:58:17.497    10250: 31784: 31784 E/LSPosed-Bridge  ] TgVlQGkRWEF.clsVYnC.Bz.XposedHelpers$InvocationTargetError: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /data/app/~~SJg0T_IaHTZQOUzHwOP5nw==/com.instagram.android-_4dHFA7-NyrnB4fWnTJ_kg==/base.apk
> 	at TgVlQGkRWEF.clsVYnC.Bz.XposedHelpers.callMethod(SourceFile:2)
> 	at com.chacha.igexperiments.Module.handleLoadPackage(Module.java:115)
> 	at TgVlQGkRWEF.clsVYnC.Bz.IXposedHookLoadPackage$Wrapper.handleLoadPackage(Unknown Source:2)
> 	at TgVlQGkRWEF.clsVYnC.Bz.callbacks.XC_LoadPackage.call(Unknown Source:6)
> 	at TgVlQGkRWEF.clsVYnC.Bz.callbacks.XCallback.callAll(Unknown Source:26)
> 	at h0.a(Unknown Source:271)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at J.callback(Unknown Source:250)
> 	at LSPHooker_.getClassLoader(Unknown Source:8)
> 	at android.app.LoadedApk.getResources(LoadedApk.java:1310)
> 	at android.app.ContextImpl.createAppContext(ContextImpl.java:3014)
> 	at android.app.ContextImpl.createAppContext(ContextImpl.java:3006)
> 	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6623)
> 	at android.app.ActivityThread.access$1500(ActivityThread.java:247)
> 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2053)
> 	at android.os.Handler.dispatchMessage(Handler.java:106)
> 	at android.os.Looper.loopOnce(Looper.java:201)
> 	at android.os.Looper.loop(Looper.java:288)
> 	at android.app.ActivityThread.main(ActivityThread.java:7839)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
> 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
> Caused by: android.content.pm.PackageParser$PackageParserException: Failed to read manifest from /data/app/~~SJg0T_IaHTZQOUzHwOP5nw==/com.instagram.android-_4dHFA7-NyrnB4fWnTJ_kg==/base.apk
> 	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1231)
> 	at android.content.pm.PackageParser.parseMonolithicPackage(PackageParser.java:1177)
> 	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1081)
> 	at android.content.pm.PackageParser.parsePackage(PackageParser.java:1090)
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	... 22 more
> Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.app.Application.getSystemService(java.lang.Class)' on a null object reference
> 	at android.content.pm.PackageParser.parseBaseApkCommon(PackageParser.java:2400)
> 	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1892)
> 	at android.content.pm.PackageParser.parseBaseApk(PackageParser.java:1215)
> 	... 26 more
> ```
> 
> here i succeed copying from log file i hope that its enough

hi what rom are you using please ?

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -7,7 +7,7 @@ android {
         minSdkVersion 24
         targetSdkVersion 34
         versionCode 1
-        versionName "2.5"
+        versionName "2.5.1"
 
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }

--- a/app/src/main/java/com/chacha/igexperiments/Module.java
+++ b/app/src/main/java/com/chacha/igexperiments/Module.java
@@ -4,14 +4,11 @@ import static de.robv.android.xposed.XposedHelpers.findAndHookMethod;
 
 import android.annotation.SuppressLint;
 import android.app.AndroidAppHelper;
-import android.content.Context;
 import android.os.Handler;
 import android.os.Looper;
 import android.os.StrictMode;
 import android.util.Log;
 import android.widget.Toast;
-
-import androidx.fragment.app.FragmentActivity;
 
 import org.json.JSONArray;
 import org.json.JSONException;
@@ -26,7 +23,6 @@ import java.util.Scanner;
 
 import de.robv.android.xposed.IXposedHookLoadPackage;
 import de.robv.android.xposed.IXposedHookZygoteInit;
-import de.robv.android.xposed.XC_MethodHook;
 import de.robv.android.xposed.XC_MethodReplacement;
 import de.robv.android.xposed.XposedBridge;
 import de.robv.android.xposed.XposedHelpers;

--- a/app/src/main/java/com/chacha/igexperiments/Module.java
+++ b/app/src/main/java/com/chacha/igexperiments/Module.java
@@ -170,30 +170,6 @@ public class Module implements IXposedHookLoadPackage, IXposedHookZygoteInit {
                 XposedBridge.log(getTime() + "End!");
             }
 
-            try {
-                XposedHelpers.findAndHookMethod("X.Dq5", lpparam.classLoader, "onClick", android.view.View.class, new XC_MethodHook() {
-                    @Override
-                    protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
-                        super.beforeHookedMethod(param);
-                        XposedHelpers.setIntField(param.thisObject, "A01", 43);
-                        XposedBridge.log("Here - WhitehatOptions");
-                        showToast("WhitehatOptionsFragment");
-                    }
-
-                    @Override
-                    protected void afterHookedMethod(MethodHookParam param) throws Throwable {
-                        super.afterHookedMethod(param);
-                        XposedHelpers.setIntField(param.thisObject, "A01", 43);
-                        XposedBridge.log("Here - WhitehatOptions");
-                        showToast("WhitehatOptionsFragment");
-
-                    }
-                });
-            } catch (Exception e) {
-                XposedBridge.log(getTime() + "Exp" + e.getMessage());
-            }
-
-
         }
 
         if (lpparam.packageName.equals(Utils.MY_PACKAGE_NAME)) {

--- a/app/src/main/java/com/chacha/igexperiments/Module.java
+++ b/app/src/main/java/com/chacha/igexperiments/Module.java
@@ -4,11 +4,14 @@ import static de.robv.android.xposed.XposedHelpers.findAndHookMethod;
 
 import android.annotation.SuppressLint;
 import android.app.AndroidAppHelper;
+import android.content.Context;
 import android.os.Handler;
 import android.os.Looper;
 import android.os.StrictMode;
 import android.util.Log;
 import android.widget.Toast;
+
+import androidx.fragment.app.FragmentActivity;
 
 import org.json.JSONArray;
 import org.json.JSONException;
@@ -23,6 +26,7 @@ import java.util.Scanner;
 
 import de.robv.android.xposed.IXposedHookLoadPackage;
 import de.robv.android.xposed.IXposedHookZygoteInit;
+import de.robv.android.xposed.XC_MethodHook;
 import de.robv.android.xposed.XC_MethodReplacement;
 import de.robv.android.xposed.XposedBridge;
 import de.robv.android.xposed.XposedHelpers;
@@ -78,6 +82,7 @@ public class Module implements IXposedHookLoadPackage, IXposedHookZygoteInit {
 
         if (lpparam.packageName.equals(Utils.IG_PACKAGE_NAME)) {
 
+
             boolean success = false;
 
             try {
@@ -88,9 +93,12 @@ public class Module implements IXposedHookLoadPackage, IXposedHookZygoteInit {
                     success = true;
                     // DEV PURPOSES
                     //showToast("HECKER MODE");
-                    //showToast("(IGExperiments) Hooking class: " + classToHook);
-                    //showToast("(IGExperiments) Hooking method: " + methodToHook);
-                    //showToast("(IGExperiments) Hooking Second class: " + secondClassToHook);
+                    //showToast("(IGExperiments) Hooking class: " + className);
+                    //showToast("(IGExperiments) Hooking method: " + methodName);
+                    //showToast("(IGExperiments) Hooking Second class: " + secondClassName);
+                    XposedBridge.log(getTime() + "(IGExperiments) Hooking class: " + className);
+                    XposedBridge.log(getTime() + "(IGExperiments) Hooking method: " + methodName);
+                    XposedBridge.log(getTime() + "(IGExperiments) Hooking Second class: " + secondClassName);
 
                     Class<?> targetClass = XposedHelpers.findClass(className, lpparam.classLoader);
                     Class<?> secondTargetClass = XposedHelpers.findClass(secondClassName, lpparam.classLoader);
@@ -103,7 +111,7 @@ public class Module implements IXposedHookLoadPackage, IXposedHookZygoteInit {
                                 }
                             });
                 }
-            } catch (Exception e) {
+            } catch (Exception ignored) {
 
             }
             // if not success it means the hecker mode wasn't used "Root wasn't granted at the first place"
@@ -144,22 +152,47 @@ public class Module implements IXposedHookLoadPackage, IXposedHookZygoteInit {
                                 }
                             });
 
-
                 } catch (InstantiationException | IllegalAccessException e) {
                     XposedBridge.log(getTime() + "Reflection error: " + e.getMessage());
                     e.printStackTrace();
                 } catch (IllegalArgumentException e) {
                     XposedBridge.log(getTime() + "Illegal argument in method call: " + e.getMessage());
                     e.printStackTrace();
-                } catch (Exception e) { // Catch other exceptions that might not be predicted
+                }
+                catch (XposedHelpers.InvocationTargetError e){
+                    showToast("Didn't work, Please use Hecker mode!");
+                    XposedBridge.log(getTime() + "Auto hook didn't work, Use Hecker mode!");
+                }
+                catch (Exception e) { // Catch other exceptions that might not be predicted
                     XposedBridge.log(getTime() + "Unhandled exception: " + e.getMessage());
                     e.printStackTrace();
                 }
                 XposedBridge.log(getTime() + "End!");
             }
 
+            try {
+                XposedHelpers.findAndHookMethod("X.Dq5", lpparam.classLoader, "onClick", android.view.View.class, new XC_MethodHook() {
+                    @Override
+                    protected void beforeHookedMethod(MethodHookParam param) throws Throwable {
+                        super.beforeHookedMethod(param);
+                        XposedHelpers.setIntField(param.thisObject, "A01", 43);
+                        XposedBridge.log("Here - WhitehatOptions");
+                        showToast("WhitehatOptionsFragment");
+                    }
 
-            //}
+                    @Override
+                    protected void afterHookedMethod(MethodHookParam param) throws Throwable {
+                        super.afterHookedMethod(param);
+                        XposedHelpers.setIntField(param.thisObject, "A01", 43);
+                        XposedBridge.log("Here - WhitehatOptions");
+                        showToast("WhitehatOptionsFragment");
+
+                    }
+                });
+            } catch (Exception e) {
+                XposedBridge.log(getTime() + "Exp" + e.getMessage());
+            }
+
 
         }
 
@@ -244,9 +277,7 @@ public class Module implements IXposedHookLoadPackage, IXposedHookZygoteInit {
         String currentTime = sdf.format(new Date());
 
         // Log the current time
-        String time = "Time: " + currentTime + " - ";
-
-        return time;
+        return "Time: " + currentTime + " - ";
     }
 
 }

--- a/classes_to_hook.json
+++ b/classes_to_hook.json
@@ -8,11 +8,11 @@
                 "download": "https://www.apkmirror.com/apk/instagram/instagram-instagram/instagram-instagram-314-0-0-20-114-release/"
         },
         {
-                "version": "314.0.0.5.114 - Beta",
-                "class_to_hook": "X.186",
+                "version": "315.0.0.21.109 - Beta",
+                "class_to_hook": "X.17I",
                 "method_to_hook": "A00",
-                "second_class_to_hook":"X.0m3",
-                "download": "https://www.apkmirror.com/apk/instagram/instagram-instagram/instagram-instagram-314-0-0-5-114-release/"
+                "second_class_to_hook":"X.0gu",
+                "download": "https://www.apkmirror.com/apk/instagram/instagram-instagram/instagram-instagram-315-0-0-21-109-release/"
         },
         {
                 "version": "316.0.0.0.42 - Alpha",

--- a/classes_to_hook.json
+++ b/classes_to_hook.json
@@ -15,11 +15,11 @@
                 "download": "https://www.apkmirror.com/apk/instagram/instagram-instagram/instagram-instagram-314-0-0-5-114-release/"
         },
         {
-                "version": "315.0.0.0.109 - Alpha",
-                "class_to_hook": "X.17I",
+                "version": "316.0.0.0.42 - Alpha",
+                "class_to_hook": "X.17H",
                 "method_to_hook": "A00",
-                "second_class_to_hook":"X.0gw",
-                "download": "https://www.apkmirror.com/apk/instagram/instagram-instagram/instagram-instagram-315-0-0-0-109-release/"
+                "second_class_to_hook":"X.0gx",
+                "download": "https://www.apkmirror.com/apk/instagram/instagram-instagram/instagram-instagram-316-0-0-0-42-release/"
         }
   ]
 }

